mcs/class/corlib/System.Runtime.Remoting.Messaging/CADMessages.cs

```diff
@@ -74,7 +74,88 @@ internal class CADMessageBase {
 		protected byte [] _serializedArgs = null;
 		protected int _propertyCount = 0;
 		protected CADArgHolder _callContext;
-		
+		internal RuntimeMethodHandle MethodHandle;
+		internal string FullTypeName;
+		internal MethodBase _method;
+
+		public CADMessageBase (IMethodMessage msg) {
+			MethodHandle = msg.MethodBase.MethodHandle;
+			FullTypeName = msg.MethodBase.DeclaringType.AssemblyQualifiedName;
+		}
+
+		internal MethodBase method {
+			get {
+				if (_method == null) {
+					_method = GetMethod();
+				}
+				return _method;
+			}
+		}
+
+		internal MethodBase GetMethod ()
```

Comment:
This big method was moved. I agree with the others, having the move in a separate commit would make it clear if anything changed in it. (did it change?)